packages/flutter/lib/src/material/time_picker.dart

```diff
@@ -1005,13 +1020,13 @@ class _DialState extends State<_Dial> with SingleTickerProviderStateMixin {
     switch (widget.mode) {
       case _TimePickerMode.hour:
         if (widget.is24h) {
-          primaryOuterLabels = _init24HourOuterRing(theme.textTheme, localizations);
-          secondaryOuterLabels = _init24HourOuterRing(theme.accentTextTheme, localizations);
-          primaryInnerLabels = _init24HourInnerRing(theme.textTheme, localizations);
-          secondaryInnerLabels = _init24HourInnerRing(theme.accentTextTheme, localizations);
+          primaryOuterLabels = _init24HourOuterRing(theme.textTheme, localizations, media.alwaysUse24HourFormat);
```

Comment:
i was a bit confused when reading this because I thought i interpreted "_initFoo" as methods that get called from initState. Maybe rename those buildFoo to make it clearer they are called every build?